cypress/integration/BounceRulesContainer/bounce_rules.spec.js

```diff
@@ -16,23 +34,24 @@ describe("Bounce Rules Page", () => {
   });
 
   it("should create a bounce rule", () => {
-    BounceRulesPage.createRuleButton.click();
-    BounceRulesPage.fillForm(
-      "1",
-      "cypressTest",
-      "528",
-      "testDescription",
-      "492",
-      "testRegex"
-    );
-    BounceRulesPage.confirmModal.should("be.visible");
-    BounceRulesPage.confirmationSubmit.click();
-    BounceRulesPage.testRule.should("be.visible");
+    BounceRulesPage.deleteBounceRuleAPI(testCreateRule).then(() => {
+      BounceRulesPage.createBounceRuleUI(testCreateRule).then(() => {
+        BounceRulesPage.confirmModal.should("be.visible");
+        BounceRulesPage.confirmationSubmit.click().then(() => {
```

Comment:
ahh I think you will be all good without chaining the .thens of the click and further assertions and we shouldn't delete again after it's created since this test is focused on validating the creation part. It's all good to delete the bounce rule through API then create bounce rule through UI and inside of the second .then() you will make the assertions on the created rule showing up (the confirmModal and confirmationSubmit steps should also be part of the createBounceRuleUI function). Let's rename the createdCreateRuleAPI and createdDeleteRuleAPI since they're not actually being created through the API but through the UI - you can rename to something more general like testBounceRuleToCreate, testBounceRuleToDelete

Reply:
We should mostly be chaining off of the higher level functions i.e. createBounceRuleUI or asynchronous stuff through API like deleteBounceRuleAPI rather than after each cypress command. clicks will for the most part happen in the order we expect but on a side note the confirmation step should also be part of the createBounceRuleUI function so  at the end of the definition will be return confirmationSubmit.click()